evmap-derive/src/lib.rs

```diff
@@ -1,10 +1,57 @@
+//! This crate provides procedural derive macros to simplify the usage of `evmap`.
+//!
+//! Currently, only `#[derive(ShallowCopy)]` is supported; see below.
+#![deny(missing_docs)]
 use proc_macro2::{Ident, TokenStream};
 use quote::{format_ident, quote, quote_spanned};
 use syn::spanned::Spanned;
 use syn::{
     parse_macro_input, parse_quote, Data, DeriveInput, Fields, GenericParam, Generics, Index,
 };
 
+/// Implementation for `#[derive(ShallowCopy)]`
+///
+/// evmap provides the [ShallowCopy](evmap::shallow_copy::ShallowCopy) trait, allowing you to cheaply alias types that don't otherwise implement `Copy`.
```

Comment:
```suggestion
/// evmap provides the [`ShallowCopy`](evmap::shallow_copy::ShallowCopy) trait, which allows you to cheaply alias types that don't otherwise implement `Copy`.
```